front/app/components/ConditionalWrapper.tsx

```diff
@@ -0,0 +1,12 @@
+import { ReactElement, ReactNode } from 'react';
+
+interface Props {
+  condition: boolean;
+  children: ReactElement;
+  wrapper: (children: ReactNode) => ReactElement;
```

Comment:
This prop name is unclear - it doesn't really describe well what this wrapper does and it's purpose.